depthai_nodes/ml/parsers/classification.py

```diff
@@ -31,20 +31,61 @@ class ClassificationParser(dai.node.ThreadedHostNode):
          An object with attributes `classes` and `scores`. `classes` is a list of classes, sorted in descending order of scores. `scores` is a list of corresponding scores.
     """
 
-    def __init__(self, classes: List[str] = None, is_softmax: bool = True):
+    def __init__(
+        self, archive_heads: list, head_name: str = "", is_softmax: bool = True
+    ):
         """Initializes the ClassificationParser node.
 
-        @param classes: List of class names to be used for linking with their respective
-            scores.
-        @param is_softmax: If False, the scores are converted to probabilities using
-            softmax function.
+        Attributes
+        ----------
+        archive_metadata : list
+            List of all archive head objects: [<depthai.nn_archive.v1.Head object>, ...]
+        head_name : str
+            If the list has multiple heads, this will specify which head to use. will throw an error if archive heads is longer then 1 and no head name is provided
         """
 
         dai.node.ThreadedHostNode.__init__(self)
         self.out = self.createOutput()
         self.input = self.createInput()
-        self.classes = classes if classes is not None else []
-        self.n_classes = len(self.classes)
+
+        if len(archive_heads) == 0:
+            raise ValueError("No heads parsed from archive")
+
+        if len(archive_heads) > 1 and head_name == "":
+            raise ValueError("Multiple heads detected, please specify head name")
+
+        self.head = archive_heads[0]
+        if head_name != "":
+            head_candidates = [
+                head
+                for head in archive_heads
+                if head.metadata.extraParams["name"] == head_name
+            ]  # possibly integrate name into dai such that we can call head.name
+            if len(head_candidates) == 0:
+                raise ValueError("Head name not found in archive")
+            if len(head_candidates) > 1:
+                raise ValueError(
+                    "Multiple heads with the same name found in archive, please specify a unique head name"
+                )
+            self.head = head_candidates[0]
+
+        if (
+            self.head.parser != "ClassificationParser"
+        ):  # Maybe we change to a more generic name like 'classification'
+            raise ValueError(
+                "Head is not a classification head, please correct the nn_archive"
+            )
+
+        layers = self.head.outputs
+        if len(layers) != 1:
+            raise ValueError(
+                "Only one output layer supported for classification, please correct nn_archive/ model"
+            )
+        self.output_layer_name = layers[0]
+
+        self.classes = self.head.metadata.classes
+        self.n_classes = self.head.metadata.nClasses
```

Comment:
should we keep it open that parser-specific parameters could be defined manually? Because as we have it set currently, it's necessary to use a NN Archive. I see how this is useful for DAI but I also see use cases when you'd only have a model and would want to define things manually. In this case, `archive_heads` parameter should be optional.

Reply:
I agree that we leave set functions so the user can change the parameters. If we make base parser class we can have a method `build` like dai-experiments host host have it. To the `build` we pass only the head-metadata all that method takes care off mapping all metadata parameters to parser's parameters.

Reply:
So the actual parser would have say `metadata: Dict` parameter, and a `.build(metadata: Union(dict, list)` that would do two things: If its a dict it would just set up class attributes if a list (we presume its a list of heads from nn archive) and it assignes class attributes as it does now?